### ISSUE TITLE
PyCharm 2016.2.1 

### ISSUE THREAD

**Issue by aotte** [vsch/idea-multimarkdown#286]
# Environment
Windows 10
PyCharm 2016.2.1 (stable release)
Markdown Navigator 2.0.0

# How to reproduce

The best way I have found to reliably reproduce it is in a plain Python project, create a new markdown file, start out with typing back ticks. The moment the third back tick is entered the event log fills up and eventually stops logging.

Type any character after the third back tick and the error goes away and the three back ticks neatly take on their syntax highlighting.

# From the Event Log

PyCharm's event log is starting to show multiple entries stating:

```StringIndexOutOfBoundsException: String index: 9914 out of range: 0, 3```

It is also reported out as an IDE fatal error with the following message:
String index: 9914 out of range: 0, 3

```java.lang.StringIndexOutOfBoundsException: String index: 9914 out of range: 0, 3
	at com.vladsch.flexmark.internal.util.sequence.SubSequence.baseSubSequence(SubSequence.java:108)
	at com.vladsch.idea.multimarkdown.parser.LexParserFlexmarkASTVisitor.visit(LexParserFlexmarkASTVisitor.java:347)
	at com.vladsch.flexmark.internal.util.ast.VisitHandler.visit(VisitHandler.java:28)
	at com.vladsch.flexmark.internal.util.ast.NodeVisitor.visit(NodeVisitor.java:55)
	at com.vladsch.flexmark.internal.util.ast.NodeVisitor.visitChildren(NodeVisitor.java:47)
	at com.vladsch.idea.multimarkdown.parser.LexParserFlexmarkASTVisitor.visitChildren(LexParserFlexmarkASTVisitor.java:133)
	at com.vladsch.idea.multimarkdown.parser.LexParserFlexmarkASTVisitor.visit(LexParserFlexmarkASTVisitor.java:138)
	at com.vladsch.flexmark.internal.util.ast.VisitHandler.visit(VisitHandler.java:28)
	at com.vladsch.flexmark.internal.util.ast.NodeVisitor.visit(NodeVisitor.java:55)
	at com.vladsch.idea.multimarkdown.parser.LexParserFlexmarkASTVisitor.build(LexParserFlexmarkASTVisitor.java:129)
	at com.vladsch.idea.multimarkdown.parser.MarkdownLexParser.parseFlexmarkMarkdown(MarkdownLexParser.java:125)
	at com.vladsch.idea.multimarkdown.parser.MultiMarkdownLexParserManager.parseFlexmark(MultiMarkdownLexParserManager.java:269)
	at com.vladsch.idea.multimarkdown.parser.MultiMarkdownLexParserManager.parseMarkdown(MultiMarkdownLexParserManager.java:201)
	at com.vladsch.idea.multimarkdown.parser.MultiMarkdownLexer.start(MultiMarkdownLexer.java:116)
	at com.intellij.lexer.Lexer.start(Lexer.java:47)
	at com.intellij.lang.impl.PsiBuilderImpl.a(PsiBuilderImpl.java:268)
	at com.intellij.lang.impl.PsiBuilderImpl.<init>(PsiBuilderImpl.java:194)
	at com.intellij.lang.impl.PsiBuilderImpl.<init>(PsiBuilderImpl.java:202)
	at com.intellij.lang.impl.PsiBuilderFactoryImpl.createBuilder(PsiBuilderFactoryImpl.java:52)
	at com.intellij.psi.tree.ILazyParseableElementType.doParseContents(ILazyParseableElementType.java:83)
	at com.intellij.psi.tree.IFileElementType.parseContents(IFileElementType.java:43)
	at com.intellij.psi.impl.source.tree.LazyParseableElement.b(LazyParseableElement.java:192)
	at com.intellij.psi.impl.source.tree.LazyParseableElement.getFirstChildNode(LazyParseableElement.java:240)
	at com.intellij.psi.impl.source.tree.LazyParseableElement.getFirstChildNode(LazyParseableElement.java:39)
	at com.intellij.psi.impl.source.text.BlockSupportImpl.a(BlockSupportImpl.java:331)
	at com.intellij.psi.impl.source.text.BlockSupportImpl.mergeTrees(BlockSupportImpl.java:277)
	at com.intellij.psi.impl.source.text.BlockSupportImpl.a(BlockSupportImpl.java:216)
	at com.intellij.psi.impl.source.text.BlockSupportImpl.reparseRange(BlockSupportImpl.java:92)
	at com.intellij.psi.impl.DocumentCommitThread.doCommit(DocumentCommitThread.java:803)
	at com.intellij.psi.impl.DocumentCommitThread$6.run(DocumentCommitThread.java:581)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1034)
	at com.intellij.psi.impl.DocumentCommitThread.a(DocumentCommitThread.java:599)
	at com.intellij.psi.impl.DocumentCommitThread.access$400(DocumentCommitThread.java:81)
	at com.intellij.psi.impl.DocumentCommitThread$2.run(DocumentCommitThread.java:415)
	at com.intellij.openapi.progress.impl.CoreProgressManager.a(CoreProgressManager.java:494)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:443)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:54)
	at com.intellij.psi.impl.DocumentCommitThread.d(DocumentCommitThread.java:412)
	at com.intellij.psi.impl.DocumentCommitThread.run(DocumentCommitThread.java:362)
	at com.intellij.util.concurrency.BoundedTaskExecutor.runFirstTaskThenPollAndRunRest(BoundedTaskExecutor.java:178)
	at com.intellij.util.concurrency.BoundedTaskExecutor.access$000(BoundedTaskExecutor.java:40)
	at com.intellij.util.concurrency.BoundedTaskExecutor$2.run(BoundedTaskExecutor.java:197)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)```

**Comment by vsch** [comment]
@aotte, thank you for the such a detailed bug report. I will post a patch fix as soon as I have it complete.

**Comment by aotte** [comment]
@vsch you are welcome. Glad to be of help.

**Comment by aotte** [comment]
It looks like 284 is the same issue as mine...

**Comment by vsch** [comment]
@aotte, patch release available from eap channel or direct download: <https://github.com/vsch/idea-multimarkdown/raw/master/dist/idea-multimarkdown.2.0.0.4.zip>

It also has issue #287 fixed. 

You can download the patch release and install plugin from disk.

**Comment by aotte** [comment]
@vsch Confirmed fixed in 2.0.0.4. Thank you.
